### PR TITLE
Use more standard rspec for shared example

### PR DESCRIPTION
**Pull request by nik9000**
This switches our single shared example to use more standard rspec,
adding contexts to make it clear what is being tested and switching from
the `convert` method to the `convert` context.

### CODE REVIEW

File: resources/asciidoctor/spec/shared_examples/does_not_break_line_numbers.rb [hunk 1]
```diff
@@ -1,19 +1,30 @@
...
-      include::resources/does_not_break_line_numbers/missing_callout.adoc[]
-    ASCIIDOC
-    convert input, {}, match(/missing_callout.adoc: line 3: no callout found for <1>/)
+    context 'when there is an error in an included file' do
+      let(:included) do
+        'resources/does_not_break_line_numbers/missing_callout.adoc'
+      end
+      let(:input) { "include::#{included}[]" }
+      let(:expected) { "WARN: #{included}: line 3: no callout found for <1>" }
+      it "reports the rigth line for the error" do
```
Comment (estolfo) [inline resources/asciidoctor/spec/shared_examples/does_not_break_line_numbers.rb]: minor type `rigth`

**Comment by nik9000** [comment]
@estolfo, I've fixed the typo!

**Comment by estolfo** [comment]
lol just realized I had a typo in my comment!

**Comment by nik9000** [comment]
> lol just realized I had a typo in my comment!

The best kind of comment!

**Pull request merged**